src/FileStorage/BucketFileStorage.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+declare(strict_types=1);
+
+namespace SprykerSdk\Integrator\FileStorage;
+
+use Aws\Credentials\Credentials;
+use Aws\S3\S3Client;
+use RuntimeException;
+use SprykerSdk\Integrator\IntegratorConfig;
+
+class BucketFileStorage implements BucketFileStorageInterface
+{
+    /**
+     * @var string
+     */
+    protected const BUCKET_KEY = 'Bucket';
+
+    /**
+     * @var string
+     */
+    protected const OBJECT_KEY = 'Key';
+
+    /**
+     * @var string
+     */
+    protected const BODY_KEY = 'Body';
+
+    /**
+     * @var string
+     */
+    protected const VERSION_KEY = 'version';
+
+    /**
+     * @var string
+     */
+    protected const LATEST_KEY = 'latest';
+
+    /**
+     * @var string
+     */
+    protected const REGION_KEY = 'region';
+
+    /**
+     * @var string
+     */
+    protected const CREDENTIALS_KEY = 'credentials';
+
+    /**
+     * @var string
+     */
+    protected const CONTENT_LENGTH_KEY = 'ContentLength';
+
+    /**
+     * @var \Aws\S3\S3Client $s3Client
+     */
+    protected ?S3Client $s3Client = null;
```

Comment:
I'd recommend extracting s3 client to the adapter class along with the `getS3Client` and `validateCredentials` methods.

Reply:
we will need to rewrite off method of the file if the S3Client changes - as a result the class is a kind of adapter 
is it suitable to rename the class to `BucketFileStorageAdapter`?

Reply:
yes. I didn't get it at the first place.